components/collector/src/base_collectors/collector.py

```diff
@@ -43,41 +40,37 @@ def record_health() -> None:
-        timeout = aiohttp.ClientTimeout(total=120)
-        while True:
-            self.record_health()
-            # The TCPConnector has limit 0, meaning unlimited, because aiohttp only closes connections when the response
-            # is closed. But due to the architecture of the collector (first collect all the responses, then parse them)
-            # the responses are kept around relatively long, and hence the connections too. To prevent time-outs while
-            # aiohttp waits for connections to become available we don't limit the connection pool size.
-            async with aiohttp.ClientSession(
-                connector=aiohttp.TCPConnector(limit=0, ssl=False),
-                raise_for_status=True,
-                timeout=timeout,
-                trust_env=True,
-            ) as session:
-                with timer() as collection_timer:
-                    await self.collect_metrics(session)
-            sleep_duration = max(0, config.MAX_SLEEP_DURATION - collection_timer.duration)
-            logging.info(
-                "Collecting took %.1f seconds. Sleeping %.1f seconds...",
-                collection_timer.duration,
-                sleep_duration,
-            )
-            await asyncio.sleep(sleep_duration)
+        # The TCPConnector has limit 0, meaning unlimited, because aiohttp only closes connections when the response
+        # is closed. But due to the architecture of the collector (first collect all the responses, then parse them)
+        # the responses are kept around relatively long, and hence the connections too. To prevent time-outs while
+        # aiohttp waits for connections to become available we don't limit the connection pool size.
+        async with aiohttp.ClientSession(
+            connector=aiohttp.TCPConnector(limit=0, ssl=False),
+            raise_for_status=True,
+            timeout=aiohttp.ClientTimeout(total=120),
+            trust_env=True,
+        ) as session:
+            while True:
+                self.__log_tasks("waking up")
+                self.record_health()
+                await self.collect_metrics(session)
+                self.__log_tasks(f"going to sleep for {config.MAX_SLEEP_DURATION} seconds", config.MEASUREMENT_LIMIT)
+                await asyncio.sleep(config.MAX_SLEEP_DURATION)
 
     async def collect_metrics(self, session: aiohttp.ClientSession) -> None:
         """Collect measurements for metrics, prioritizing edited metrics."""
         reports = get_reports(self.database)
         metrics = get_metrics_from_reports(reports)
         next_fetch = datetime.now(tz=UTC) + timedelta(seconds=config.MEASUREMENT_FREQUENCY)
-        tasks: list[Coroutine] = []
+        created_tasks = 0
         for metric_uuid, metric in self.__sorted_by_edit_status(cast(JSONDict, metrics)):
-            if len(tasks) >= config.MEASUREMENT_LIMIT:
-                break
             if self.__should_collect(metric_uuid, metric):
```

Comment:
Why not change this into `if not should collect: continue` and skip a level of indentation?